client/register.js

```diff
@@ -1,27 +1,72 @@
 import jQuery from "jquery";
 
 (($) => {
+  let registrationForm = document.getElementById("register-form");
   const setFormError = (message) => {
     $("#register-form-error").text(message);
     $("#register-form-error").show();
   };
 
-  $("#register-form").on("submit", async (e) => {
+  registrationForm.addEventListener("submit", async (e) => {
     const username = $("#username-input").val();
     const email = $("#email-input").val();
     const password = $("#password-input").val();
     const confirmPassword = $("#confirm-password-input").val();
-
+    let errors = [];
     if (!email || !password || !username) {
       e.preventDefault();
-      setFormError("Please enter username, email, and password");
-      return;
+      errors.push("Please enter username, email, and password");
```

Comment:
Instead of writing this password validation here and then eventually copying it into the server side logic, modify validation.js, specifically passwordSchema on line 17. 

For example, adding this 
```ts
  .regex(/[a-z]/, {
    message: "Password must contain at least one lowercase letter",
  })
  .regex(/[A-Z]/, {
    message: "Password must contain at least one uppercase letter",
  })
  .regex(/[0-9]/, {
    message: "Password must contain at least one number",
  })
  .regex(/[^a-zA-Z0-9]/, {
    message: "Password must contain at least one special character",
  })
```

will run the password through each of the regular expressions, and if there is no match, it will fail validation and use the message specified.